webrender/src/display_list_flattener.rs

```diff
@@ -539,6 +543,16 @@ impl<'a> DisplayListFlattener<'a> {
             ),
         );
 
+        let clip_chain_index = match clip_and_scroll_ids.clip_node_id {
```

Comment:
Here I think you always want to use the newly added clip node which is represented by `info.clip_id`. This ensures that contents don't escape the iframe at all. Trying to reach for the parent means that the content might escape the iframe and only be contained by whatever clip happens to be the parent here. If Gecko and Servo want to contain the content to the iframe (which I think they always do), they would end up needing to push an extra clip node. I think that `info.clip_id` is the correct choice here to ensure consistent behavior.

In the future maybe it makes sense that iframes don't automatically clip, in which case we wouldn't push this clip node and we would in that case look at the clipping node of `clip_and_scroll_ids` in this spot. That would allow clients to allow iframe contents to escape their rectangles, while still avoiding having to push an extra clip node to prevent it.

Reply:
I agree that Gecko at least always wants to contain stuff inside the iframe to the iframe clip. If I'm understanding this right, what @mrobinson is suggesting would fix https://bugzilla.mozilla.org/show_bug.cgi?id=1473940